src/modules/map/components/cesium/CesiumMap.vue

```diff
@@ -1,612 +1,174 @@
-                            heading: CesiumMath.toRadians(this.cameraPosition.heading),
-                            pitch: CesiumMath.toRadians(this.cameraPosition.pitch),
-                            roll: CesiumMath.toRadians(this.cameraPosition.roll),
-                        },
-                        duration: 1,
-                    })
-                }
-            } catch (error) {
-                log.error('Error while moving the camera', error, this.cameraPosition)
-            }
-        },
-        onCameraMoveEnd() {
-            const camera = this.viewer.camera
-            const position = camera.positionCartographic
-            const cameraPosition = {
-                x: parseFloat(CesiumMath.toDegrees(position.longitude).toFixed(6)),
-                y: parseFloat(CesiumMath.toDegrees(position.latitude).toFixed(6)),
-                z: parseFloat(position.height.toFixed(1)),
-                // Wrap degrees, cesium might return 360, which is internally wrapped to 0 in
-                // store.
-                heading: wrapDegrees(parseFloat(CesiumMath.toDegrees(camera.heading).toFixed(0))),
-                pitch: wrapDegrees(parseFloat(CesiumMath.toDegrees(camera.pitch).toFixed(0))),
-                roll: wrapDegrees(parseFloat(CesiumMath.toDegrees(camera.roll).toFixed(0))),
-            }
-            if (!isEqual(cameraPosition, this.cameraPosition)) {
-                this.setCameraPosition({
-                    position: cameraPosition,
-                    ...dispatcher,
-                })
-            }
-        },
-        getCoordinateAtScreenCoordinate(x, y) {
-            const cartesian = this.viewer?.scene.pickPosition(new Cartesian2(x, y))
-            let coordinates = []
-            if (cartesian) {
-                const cartCoords = Cartographic.fromCartesian(cartesian)
-                coordinates = proj4(WGS84.epsg, this.projection.epsg, [
-                    (cartCoords.longitude * 180) / Math.PI,
-                    (cartCoords.latitude * 180) / Math.PI,
-                ])
-            } else {
-                log.error('no coordinate found at this screen coordinates', [x, y])
-            }
-            return coordinates
-        },
-        onClick(event) {
-            unhighlightGroup(this.viewer)
-            const features = []
-            let coordinates = this.getCoordinateAtScreenCoordinate(
-                event.position.x,
-                event.position.y
-            )
+<template>
+    <div
+        id="cesium"
+        ref="viewerElement"
+        class="cesium-map"
+        data-cy="cesium-map"
+        @contextmenu.prevent
+    >
+        <CesiumInteractions v-if="viewerCreated" />
+        <CesiumCamera v-if="viewerCreated" />
+        <CesiumBackgroundLayer v-if="viewerCreated" />
+        <CesiumVisibleLayers v-if="viewerCreated" />
+        <CesiumHighlightedFeatures v-if="viewerCreated" />
+        <CesiumGeolocationFeedback v-if="viewerCreated" />
```

Comment:
If the `<div>` on which the Cesium viewer will attach itself is not present, the whole thing crashes down. That's why I had to check each sub-elements instead of being able to `v-if` the whole block.